pkg/util/fs_util.go

```diff
@@ -677,3 +672,49 @@ func CreateTargetTarfile(tarpath string) (*os.File, error) {
 	return os.Create(tarpath)
 
 }
+
+// Returns true if a file is a symlink
+func IsSymlink(fi os.FileInfo) bool {
+	return fi.Mode()&os.ModeSymlink != 0
+}
+
+var ErrNotSymLink = fmt.Errorf("not a symlink")
+
+func CanonicalizeLink(path string) (string, error) {
+	fi, err := os.Lstat(path)
+	if err != nil {
+		return "", err
+	}
+	if !IsSymlink(fi) {
+		return "", ErrNotSymLink
+	}
+	return filepath.EvalSymlinks(path)
```

Comment:
iirc `EvalSymlinks` won't raise an error if you pass it a resolved path. Is there a need to check if the path is a symlink? Is the error handling used as control flow somewhere else to determine whether a path is a symlink?

Reply:
this is now changed. Please take a look again.